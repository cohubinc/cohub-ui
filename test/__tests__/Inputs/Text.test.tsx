import React from "react";
import { mount } from "enzyme";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Inputs } from "dist";

describe("Text Input", () => {
  it("renders without crashing", async () => {
    const wrapper = mount(<Inputs.Text />);

    expect(wrapper).toMatchSnapshot();
  });

  it("receives text", () => {
    const { getByLabelText } = render(
      <Inputs.Text data-qa="textInputId" label="Text Label" id="text_input" />
    );

    expect(getByLabelText("Text Label")).toBeEmpty();

    userEvent.type(getByLabelText("Text Label"), "Hello, World!");

    expect(getByLabelText("Text Label")).toHaveValue("Hello, World!");
  });
});
