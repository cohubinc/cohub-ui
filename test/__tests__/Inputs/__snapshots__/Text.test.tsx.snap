// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input renders without crashing 1`] = `
<Text$1>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Base
      autoComplete="off"
      autoFocus={false}
      data-qa="base-input-element"
      data-qa-label="base-input-element-label"
      error={false}
      type="text"
    >
      <FloatingLabelWrapper
        data-qa-label="base-input-element-label"
        error={false}
      >
        <div
          className="FloatingLabelWrapper GenericInput "
          style={
            Object {
              "cursor": "text",
              "position": "relative",
            }
          }
        >
          <div
            className="inputWrapper   bd-radius"
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              data-qa="base-input-element"
              data-qa-label="base-input-element-label"
              onBlur={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "color": "hsl(0, 2%, 12%)",
                  "cursor": "text",
                  "textAlign": undefined,
                }
              }
              type="text"
            />
            <span
              className="bar "
            />
          </div>
        </div>
      </FloatingLabelWrapper>
    </Base>
  </div>
</Text$1>
`;
